app/scripts/controllers/incoming-transactions.test.js

```diff
@@ -1343,4 +1344,89 @@ describe('IncomingTransactionsController', function () {
       });
     });
   });
+
+  describe('_allowedToMakeFetchIncomingTx', function () {
```

Comment:
Nit: Generally I advise avoiding testing internal functions. This can pose an obstacle to refactoring, and it makes the tests farther from the API and real-world usage of the module.

Framing the functionality in _external_ terms generally makes the purpose of each test easier to grasp, _unblocks_ refactors rather than adding friction to them. Centering on external usage can be helpful in highlighting awkward API decisions too (i.e. if the test is too hard to write, sometimes it's because the API has room for improvement, or because the path you're testing is a dead path, etc.)

I can see why you'd want to go that route here to avoid the repetition. But repetition in a test isn't so bad. We can make things more concise by moving things into helper functions as well, potentially.

Not a blocker though; we test private functions all over the place today, it's just usually not ideal.